README.md
# ThesisPlots
[![status: archive](https://github.com/GIScience/badges/raw/master/status/archive.svg)](https://github.com/GIScience/badges#archive)

This package has been archived.

--------------

Collection of plots of dissertation thesis

[![Coverage Status](https://coveralls.io/repos/github/JoranAngevaare/thesis_plots/badge.svg?branch=master)](https://coveralls.io/github/JoranAngevaare/thesis_plots?branch=master)
[![CodeFactor](https://www.codefactor.io/repository/github/joranangevaare/thesis_plots/badge)](https://www.codefactor.io/repository/github/joranangevaare/thesis_plots)
[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.6527303.svg)](https://doi.org/10.5281/zenodo.6527303)


HISTORY.md
0.2.0 / 2022-10-07
------------------

## What's Changed
* Detection methods plot by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/26
* Detection principles by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/27
* Bump pytest-cov from 3.0.0 to 4.0.0 in /extra_requirements by @dependabot in https://github.com/JoranAngevaare/thesis_plots/pull/28
* Bump coverage from 6.4.4 to 6.5.0 in /extra_requirements by @dependabot in https://github.com/JoranAngevaare/thesis_plots/pull/29
* Recoil rates by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/30
* Bump nbconvert from 7.0.0 to 7.1.0 in /extra_requirements by @dependabot in https://github.com/JoranAngevaare/thesis_plots/pull/34
* Bump pytest-cov from 3.0.0 to 4.0.0 in /extra_requirements by @dependabot in https://github.com/JoranAngevaare/thesis_plots/pull/36
* Limit setting by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/39


**Full Changelog**: https://github.com/JoranAngevaare/thesis_plots/compare/v0.1.3...v0.2.0


0.1.3 / 2022-09-27
------------------

## What's Changed

* Update HISTORY.md by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/12
* Add zenodo badge by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/13
* new version of mw plots by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/14
* Bump actions/setup-python from 3 to 4 by @dependabot in https://github.com/JoranAngevaare/thesis_plots/pull/15
* Update requirements-tests.txt by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/16
* Add density_profile_plot by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/17
* Bump pytest from 7.1.2 to 7.1.3 in /extra_requirements by @dependabot
  in https://github.com/JoranAngevaare/thesis_plots/pull/24
* Bump astropy from 5.0.4 to 5.1 in /extra_requirements by @dependabot
  in https://github.com/JoranAngevaare/thesis_plots/pull/21
* Bump galpy from 1.7.2 to 1.8.0 in /extra_requirements by @dependabot
  in https://github.com/JoranAngevaare/thesis_plots/pull/19
* Bump nbconvert from 6.5.4 to 7.0.0 in /extra_requirements by @dependabot
  in https://github.com/JoranAngevaare/thesis_plots/pull/18
* Bump matplotlib from 3.5.2 to 3.6.0 in /extra_requirements by @dependabot
  in https://github.com/JoranAngevaare/thesis_plots/pull/25
* Bump numpy from 1.22.4 to 1.23.3 in /extra_requirements by @dependabot
  in https://github.com/JoranAngevaare/thesis_plots/pull/22
* Bump coverage from 6.3.2 to 6.4.4 in /extra_requirements by @dependabot
  in https://github.com/JoranAngevaare/thesis_plots/pull/23

## New Contributors

* @dependabot made their first contribution in https://github.com/JoranAngevaare/thesis_plots/pull/15

**Full Changelog**: https://github.com/JoranAngevaare/thesis_plots/compare/v0.1.2...v0.1.3

0.1.2 / 2022-05-07
------------------

## What's Changed

* Make figure prettier by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/11

0.1.1 / 2022-05-05
------------------

## What's Changed

* delete pypi by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/4
* Rot curve by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/5
* fix by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/9
* Add galaxy plot by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/7

## New Contributors

* @sourcery-ai made their first contribution in https://github.com/JoranAngevaare/thesis_plots/pull/6

**Full Changelog**: https://github.com/JoranAngevaare/thesis_plots/compare/v0.1.0...v0.1.1

**Full Changelog**: https://github.com/JoranAngevaare/thesis_plots/compare/v0.1.1...v0.1.2


0.1.0 / 2022-05-04
------------------

## What's Changed

* Make figure prettier by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/11

**Full Changelog**: https://github.com/JoranAngevaare/thesis_plots/compare/v0.1.1...v0.1.2

0.1.1 / 2022-05-04
------------------

## What's Changed

* delete pypi by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/4
* Rot curve by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/5
* fix by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/9
* Add galaxy plot by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/7

## New Contributors

* @sourcery-ai made their first contribution in https://github.com/JoranAngevaare/thesis_plots/pull/6

**Full Changelog**: https://github.com/JoranAngevaare/thesis_plots/compare/v0.1.0...v0.1.1

0.1.0 / 2022-05-04
------------------

## What's Changed

* First commits by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/1
* Update requirements-tests.txt by @JoranAngevaare in https://github.com/JoranAngevaare/thesis_plots/pull/3

## New Contributors

* @JoranAngevaare made their first contribution in https://github.com/JoranAngevaare/thesis_plots/pull/1

**Full Changelog**: https://github.com/JoranAngevaare/thesis_plots/commits/v0.1.0

0.0.0 / 2022-05-04
------------------
First commits


thesis_plots/main.py
import os
import socket
import sys
import typing as ty
from collections import defaultdict
from importlib import import_module
from platform import python_version

import matplotlib
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
from git import Repo, InvalidGitRepositoryError

root_folder = os.path.join(os.path.split(os.path.realpath(__file__))[0], '..')


def setup_plt(use_tex=True):
    """Change the plots to have my style and fashion"""
    params = {'axes.grid': False,
              #           'font.size': 16,
              #           'axes.titlesize': 22,
              #           'axes.labelsize': 18,
              #           'axes.linewidth': 2,
              #           'xtick.labelsize': 16,
              #           'ytick.labelsize': 16,
              'font.size': 20,
              'axes.titlesize': 22,
              'axes.labelsize': 20,
              'axes.linewidth': 2,
              'xtick.labelsize': 20,
              'ytick.labelsize': 20,
              'ytick.major.size': 8,
              'ytick.minor.size': 4,
              'xtick.major.size': 8,
              'xtick.minor.size': 4,
              'xtick.major.width': 2,
              'xtick.minor.width': 2,
              'ytick.major.width': 2,
              'ytick.minor.width': 2,
              'xtick.direction': 'in',
              'ytick.direction': 'in',
              'legend.fontsize': 20,
              'figure.facecolor': 'w',
              'figure.figsize': (10, 8),
              'image.cmap': 'viridis',
              'lines.linewidth': 2,
              }
    plt.rcParams.update(params)
    if use_tex and not os.environ.get('DISABLE_LATEX', False):
        # Allow latex to be disabled from the environment coverage see #30
        matplotlib.rc('text', usetex=True)


def save_fig(name,
             file_types=('png', 'pdf'),
             **kwargs):
    """Save a figure in the figures dir"""
    kwargs.setdefault('dpi', 150)
    kwargs.setdefault('bbox_inches', "tight")
    for file_type in file_types:
        path = os.path.join(root_folder, 'figures', f'{name}.{file_type}')
        plt.savefig(path, **kwargs)


def print_versions(
        modules=('thesis_plots',),
        print_output=True,
        include_python=True,
        return_string=False,
        include_git=True
):
    """
    Print versions of modules installed.

    :param modules: Modules to print, should be str, tuple or list. E.g.
        print_versions(modules=('numpy', 'thesis_plots',))
    :param return_string: optional. Instead of printing the message,
        return a string
    :param include_git: Include the current branch and latest
        commit hash
    :return: optional, the message that would have been printed
    """
    versions = defaultdict(list)
    if include_python:
        versions['module'] = ['python']
        versions['version'] = [python_version()]
        versions['path'] = [sys.executable]
        versions['git'] = [None]
    for m in to_str_tuple(modules):
        result = _version_info_for_module(m, include_git=include_git)
        if result is None: continue
        version, path, git_info = result
        versions['module'].append(m)
        versions['version'].append(version)
        versions['path'].append(path)
        versions['git'].append(git_info)
    df = pd.DataFrame(versions)
    info = f'Host {socket.getfqdn()}\n{df.to_string(index=False)}'
    if print_output:
        print(info)
    return info if return_string else df


def _version_info_for_module(module_name, include_git):
    try:
        mod = import_module(module_name)
    except ImportError:
        print(f'{module_name} is not installed')
        return
    git = None
    version = mod.__dict__.get('__version__', None)
    module_path = mod.__dict__.get('__path__', [None])[0]
    if include_git:
        try:
            repo = Repo(module_path, search_parent_directories=True)
        except InvalidGitRepositoryError:
            # not a git repo
            pass
        else:
            try:
                branch = repo.active_branch
            except TypeError:
                branch = 'unknown'
            try:
                commit_hash = repo.head.object.hexsha
            except TypeError:
                commit_hash = 'unknown'
            git = f'branch:{branch} | {commit_hash[:7]}'
    return version, module_path, git


def to_str_tuple(x: ty.Union[str, bytes, list, tuple, pd.Series, np.ndarray]) -> ty.Tuple[str]:
    """
    Convert any sensible instance to a tuple of strings
    stolen from
    https://github.com/AxFoundation/strax/blob/d3608efc77acd52e1d5a208c3092b6b45b27a6e2/strax/utils.py#242
    """
    if isinstance(x, (str, bytes)):
        return x,
    elif isinstance(x, list):
        return tuple(x)
    raise TypeError(f"Expected string or tuple of strings, got {type(x)}")


def string_to_mathrm(string):
    """wrap a string in mathrm mode for latex labels"""
    string = string.replace(' ', '\ ')
    return f'$\mathrm{{{string}}}$'
